presto-main/src/main/java/com/facebook/presto/server/InternalHttpClientConfigurationModule.java

```diff
@@ -33,41 +30,14 @@
     protected void setup(Binder binder)
     {
         InternalHttpClientConfiguration configuration = buildConfigObject(InternalHttpClientConfiguration.class);
-        binder.bindListener(new HttpClientConfigMatcher(), new ConfigurationListener(configuration));
-    }
-
-    private static class HttpClientConfigMatcher
-            extends AbstractMatcher<Binding<?>>
-    {
-        @Override
-        public boolean matches(Binding<?> binding)
-        {
-            return binding.getKey().getTypeLiteral().getRawType().equals(HttpClientConfig.class);
-        }
-    }
-
-    private static class ConfigurationListener
-            implements ProvisionListener
-    {
-        private final InternalHttpClientConfiguration configuration;
-
-        private ConfigurationListener(InternalHttpClientConfiguration configuration)
-        {
-            this.configuration = configuration;
-        }
-
-        @Override
-        public <T> void onProvision(ProvisionInvocation<T> provision)
-        {
-            HttpClientConfig config = checkType(provision.provision(), HttpClientConfig.class, "config");
-            checkState(config.getKeyStorePath() == null);
-            checkState(config.getKeyStorePassword() == null);
+        HttpClientDefaultsBinder httpClientDefaultsBinder = httpClientDefaultsBinder(binder);
```

Comment:
This failure is intermittent. Unfortunately there is no other way to register the basic authentication filter to all of the internal HTTP clients. I know what the consequences this introduce. And if there was a way to do that without modifying Airlift - i would definitely do so.